app/src/main/java/org/mozilla/fenix/compose/cfr/CFRPopupShape.kt

```diff
@@ -0,0 +1,252 @@
+
+                            val indicatorStartPx = size.width - indicatorArrowStartOffsetPx - indicatorArrowBasePx
+                            lineTo(indicatorStartPx, messageBodyHeightPx)
+                            lineTo(indicatorStartPx + indicatorArrowBasePx / 2, size.height)
+                            lineTo(indicatorStartPx + indicatorArrowBasePx, messageBodyHeightPx)
+
+                            lineTo(size.width - indicatorCornerRadiusPx, messageBodyHeightPx)
+                            quadraticBezierTo(
+                                size.width, messageBodyHeightPx,
+                                size.width, messageBodyHeightPx - indicatorCornerRadiusPx
+                            )
+                        }
+
+                        lineTo(size.width, cornerRadiusPx)
+                        quadraticBezierTo(
+                            size.width, 0f,
+                            size.width - cornerRadiusPx, 0f
+                        )
+
+                        lineTo(cornerRadiusPx, 0f)
+                        quadraticBezierTo(
+                            0f, 0f,
+                            0f, cornerRadiusPx
+                        )
+
+                        close()
+                    }
+                )
+            }
+        }
+    }
+
+    companion object {
+        /**
+         * This [Shape]'s arrow indicator will have an automatic width depending on the set height.
+         * This method allows knowing what the base width will be before instantiating the class.
+         */
+        fun getIndicatorBaseWidthForHeight(height: Int): Int {
+            return (height * INDICATOR_BASE_TO_HEIGHT_RATIO).roundToInt()
+        }
+    }
+}
+
+@Composable
+@Preview(locale = "en", name = "LTR")
+@Preview(locale = "ar", name = "RTL")
+@Preview(uiMode = Configuration.UI_MODE_NIGHT_YES, name = "Night mode")
+@Preview(uiMode = Configuration.UI_MODE_NIGHT_NO, name = "Light mode")
+private fun CFRPopupBelowShapePreview() {
+    FirefoxTheme {
+        Box(
+            modifier = Modifier
+                .height(100.dp)
+                .width(200.dp)
+                .background(
+                    shape = CFRPopupShape(IndicatorDirection.UP, 10.dp, 10.dp, 10.dp),
+                    brush = Brush.linearGradient(
+                        colors = listOf(
+                            FirefoxTheme.colors.gradientStart,
+                            FirefoxTheme.colors.gradientEnd
+                        ),
+                        end = Offset(0f, Float.POSITIVE_INFINITY),
+                        start = Offset(Float.POSITIVE_INFINITY, 0f)
+                    )
+                ),
+            contentAlignment = Alignment.Center
+        ) {
+            Text(
+                text = "This is just a test",
+                color = FirefoxTheme.colors.textOnColorPrimary
+            )
+        }
+    }
+}
+
+@Composable
+@Preview(locale = "en", name = "LTR")
+@Preview(locale = "ar", name = "RTL")
```

Comment:
nit: do we need additional previews for the Shape for RTL and LTR?

Reply:
I do think so. Would sort of us these previews as screenshot tests.
Given that the shapes are drawn programatically and there is different code to draw LRT and RTL these previews do help for a quick validation.